pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -0,0 +1,683 @@
+		Annotations:               etcd.Spec.Annotations,
+		Labels:                    etcd.Spec.Labels,
+		BackupImage:               *etcd.Spec.Backup.Image,
+		EtcdImage:                 *etcd.Spec.Etcd.Image,
+		PriorityClassName:         etcd.Spec.PriorityClassName,
+		ServiceName:               utils.GetPeerServiceName(etcd),
+		ServiceAccountName:        utils.GetServiceAccountName(etcd),
+		Affinity:                  etcd.Spec.SchedulingConstraints.Affinity,
+		TopologySpreadConstraints: etcd.Spec.SchedulingConstraints.TopologySpreadConstraints,
+
+		EtcdResources:   etcd.Spec.Etcd.Resources,
+		BackupResources: etcd.Spec.Backup.Resources,
+
+		VolumeClaimTemplateName: volumeClaimTemplateName,
+
+		FullSnapLeaseName:  utils.GetFullSnapshotLeaseName(etcd),
+		DeltaSnapLeaseName: utils.GetDeltaSnapshotLeaseName(etcd),
+
+		StorageCapacity: etcd.Spec.StorageCapacity,
+		StorageClass:    etcd.Spec.StorageClass,
+
+		ClientUrlTLS: etcd.Spec.Etcd.ClientUrlTLS,
+		PeerUrlTLS:   etcd.Spec.Etcd.PeerUrlTLS,
+		BackupTLS:    etcd.Spec.Backup.TLS,
+
+		LeaderElection: etcd.Spec.Backup.LeaderElection,
+
+		BackupStore:     etcd.Spec.Backup.Store,
+		EnableProfiling: etcd.Spec.Backup.EnableProfiling,
+
+		DeltaSnapshotPeriod:      etcd.Spec.Backup.DeltaSnapshotPeriod,
+		DeltaSnapshotMemoryLimit: etcd.Spec.Backup.DeltaSnapshotMemoryLimit,
+
+		DefragmentationSchedule: etcd.Spec.Etcd.DefragmentationSchedule,
+		FullSnapshotSchedule:    etcd.Spec.Backup.FullSnapshotSchedule,
+
+		EtcdSnapshotTimeout: etcd.Spec.Backup.EtcdSnapshotTimeout,
+		EtcdDefragTimeout:   etcd.Spec.Etcd.EtcdDefragTimeout,
+
+		GarbageCollectionPolicy: etcd.Spec.Backup.GarbageCollectionPolicy,
+		GarbageCollectionPeriod: etcd.Spec.Backup.GarbageCollectionPeriod,
+
+		SnapshotCompression: etcd.Spec.Backup.SnapshotCompression,
+		HeartbeatDuration:   etcd.Spec.Etcd.HeartbeatDuration,
+
+		Metrics:           etcd.Spec.Etcd.Metrics,
+		Quota:             etcd.Spec.Etcd.Quota,
+		ClientServiceName: utils.GetClientServiceName(etcd),
+		ClientPort:        etcd.Spec.Etcd.ClientPort,
+		PeerServiceName:   utils.GetPeerServiceName(etcd),
+		ServerPort:        etcd.Spec.Etcd.ServerPort,
+		BackupPort:        etcd.Spec.Backup.Port,
+
+		OwnerCheck: etcd.Spec.Backup.OwnerCheck,
+
+		AutoCompactionMode:      etcd.Spec.Common.AutoCompactionMode,
+		AutoCompactionRetention: etcd.Spec.Common.AutoCompactionRetention,
+		ConfigMapName:           utils.GetConfigmapName(etcd),
+	}
+
+	values.EtcdCommand = getEtcdCommand()
+	values.ReadinessProbeCommand = getReadinessProbeCommand(values)
+	values.LivenessProbCommand = getLivenessProbeCommand(values)
+	values.EtcdBackupCommand = getEtcdBackupCommand(values)
+
+	return values
+}
+
+func getEtcdCommand() []string {
+	command := []string{"" + "/var/etcd/bin/bootstrap.sh"}
+
+	return command
+}
+
+func getReadinessProbeCommand(val Values) []string {
+	command := []string{"" + "/usr/bin/curl"}
+
+	if val.ClientUrlTLS != nil {
```

Comment:
Since the URL points to either etcd server or backup-restore server based on number of replicas, the first decision point should be the replicas, and next decision point should be for whether TLS is enabled on the corresponding server, ie, if replicas==1, then check for whether `backup.tls` is present, else check for whether `etcd.clientUrlTls` is present. Please re-structure this function accordingly.

Reply:
Made the necessary changes